DEFCON Z/Assets/Tests/PlayMode Tests/MainMenuTest.cs

```diff
@@ -1,56 +1,116 @@
-﻿using System.Collections;
-using System.Collections.Generic;
+﻿using DefconZ;
 using NUnit.Framework;
 using UnityEngine;
-//using UnityEngine.TestTools;
-using DefconZ.Simulation;
-using DefconZ;
-using UnityEngine.UI;
 
 namespace Tests
 {
+    /// <summary>
+    /// Test Suite for Main Menu.
+    /// </summary>
     public class MainMenuTest
     {
+        private GameObject MainMenuGameObject = new GameObject();
+        private MainMenu MainMenu;
+
+        /// <summary>
+        /// Setup test data for each test.
+        /// </summary>
+        [SetUp]
+        public void TestInit()
+        {
+            MainMenuGameObject = new GameObject();
+
+            // Object is disabled to prevent init code being called.
+            MainMenuGameObject.SetActive(false);
+
+            // Set up the components.
+            MainMenu = MainMenuGameObject.AddComponent<MainMenu>();
+            MainMenu.settingsPanel = new GameObject();
+            MainMenu.menuPanel = new GameObject();
+
+            // Reactivate the object to init the components.
+            MainMenuGameObject.SetActive(true);
+        }
+
+        /// <summary>
+        /// Clean up data for each test.
+        /// </summary>
+        [TearDown]
+        public void TestCleanup()
+        {
+            MainMenuGameObject = null;
+            MainMenu = null;
+        }
+
+        /// <summary>
+        /// Tests that ensure MainMenu initializes panels state correctly.
+        /// </summary>
         [Test]
-        public void TestMenuSettingsPanel()
+        public void MainMenu_Should_InitializePanels_Correctly()
         {
-            // Arrange 
-            MainMenu menuTest = new MainMenu();
-            menuTest.settingsPanel = new GameObject();
-            menuTest.menuPanel = new GameObject();
-            menuTest.settings = false;
-
-            //Act
-            menuTest.ToggleSettings();
-            Assert.That(menuTest.menuPanel.activeSelf, Is.EqualTo(false));
-            Assert.That(menuTest.menuPanel.activeSelf, Is.EqualTo(true));
+            // Arrange
+            bool expectedMenuPanelState = true;
+            bool expectedSettingsPanelState = false;
+
+            // Act
+            // This part has been initiated in the Setup process.
+
+            // Assert
+            Assert.That(MainMenu.menuPanel.activeSelf, Is.EqualTo(expectedMenuPanelState));
```

Comment:
Second Assert should be moved to a separate test.
Without using multiple assert, the second Assert (should) never be able to be tested.

Reply:
I'll create two new tests. First test will test whether menu panel state should be in the correct state and another one tests whether main settings panel is at the correct state when first initialized?

Reply:
That sounds about right yeah. One should be true the other false. Could always Assert that (menu && !settings) too.

Reply:
Okay sweet, thank you.